spec/myapp_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'spec_helper'
+require_relative '../app/models/shop'
+
+describe 'myapp' do
+  include Rack::Test::Methods
+
+  def app
+    Sinatra::Application.new
+  end
+
+  describe 'root path' do
+    before do
+      @shop = Shop.new({name: 'mandy', description: 'bla'})
```

Comment:
please replace the `@shop` instance var with `let` block and you can use another `let` block to get the shop id returned by the `create` method.